hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/scope/HandleHederaOperations.java

```diff
@@ -167,6 +195,33 @@ public void createContract(
         throw new AssertionError("Not implemented");
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void createContract(
+            final long number,
+            @NonNull final ContractCreateTransactionBody body,
+            final long nonce,
+            @Nullable final Bytes evmAddress) {
+        // Create the contract account by dispatching a synthetic HAPI transaction
+        final var contractId = ContractID.newBuilder().contractNum(number).build();
+        final var synthAccountCreation = accountCreationFor(contractId, evmAddress, requireNonNull(body));
+        final var synthTxn = TransactionBody.newBuilder()
+                .cryptoCreateAccount(synthAccountCreation)
+                .build();
+        final var childRecordBuilder = context.dispatchChildTransaction(synthTxn, CryptoCreateRecordBuilder.class);
```

Comment:
Does this child record builder need to be cast to a `ContractCreateRecordBuilder` in order to create the same type of child record as in the mono service or maybe I'm wrong.

Reply:
I'm not quite sure how records will play out here...am just focusing on getting state changes working atm. 🤔